### ISSUE TITLE
setElementData will resync tables even if their values match - CLuaArgument operator== bug

### ISSUE THREAD

**Issue by Pirulax** [multitheftauto/mtasa-blue#1786]
**Describe the bug**
`setElementData` is "optimized" in a way that (in theory) prevents resyncing it the value set matches its old value, aka:
https://github.com/multitheftauto/mtasa-blue/blob/e9054698d164c1f2e205fff9f982294370e59085/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L885
In reality there's a logic error in `CLuaArgument`:
https://github.com/multitheftauto/mtasa-blue/blob/e9054698d164c1f2e205fff9f982294370e59085/Server/mods/deathmatch/logic/lua/CLuaArgument.cpp#L156-L176
The lines:
```cpp
if (pKnownTables->find(m_pTableData) == pKnownTables->end())
{
    pKnownTables->insert(m_pTableData);
    if (*iter != *iterCompare)
        return false;
}
```
Are the problem. As you can('t) see, an empty set is passed to this function. We check if `m_pTableData` (lhs CLuaArgument) is in the `pKnownTables` set (pKnownTables should hold tables that have been checked, but not in this case). Now, in the first iteration its not, obviously, but then, we insert it, and check if the first 2 values match in the table. If they do we continue the iteration, but heres the catch: we will never compare the values again, because `pKnownTables->find(m_pTableData) == pKnownTables->end()` isn't true anymore.
So, basically this function returns true, if(in case of tables):
- lhs and rhs table sizes match
- their first values match

What happened here probably is that, this function was recursing (with self referencing tables), someone added this check, which doesnt make sense, but it worked, the function didnt recurse anymore, and it held true for tables of the same size, and first value.

**To reproduce**
```lua
local t = {true, {}}
setElementData(root, "test", t) -- returns true
setElementData(root, "test", t) -- returns true, should return false
```

```
start runcode
run t = {true, {}}
run setElementData(root, "test", t)
run setElementData(root, "test", t)
```
